include/npcomp/Dialect/Basicpy/IR/BasicpyDialect.td

```diff
@@ -96,6 +96,36 @@ def Basicpy_UnknownType : DialectType<Basicpy_Dialect,
   }];
 }
 
+def Basicpy_ListType : DialectType<Basicpy_Dialect,
+    CPred<"$_self.isa<::mlir::NPCOMP::Basicpy::ListType>()">,
+    "List type"> {
+  let typeDescription = [{
+    A Python list type. In its generic form, there are no constraints on the
+    element type or length; however, it can be refined to include such
+    constraints.
```

Comment:
I feel like it would be nice to include some verbiage in both list and tuple types to clarify their semantics enough to at least distinguish them from the docs here. (e.g. mutable vs immutable)